src/main/java/apoc/atomic/Atomic.java

```diff
@@ -0,0 +1,213 @@
+	public Stream<AtomicResults> concat(@Name("container") Object container, @Name("propertyName") String property, @Name("string") String string){
+		checkIsPropertyContainer(container);
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		checkPropertyExists(propertyContainer,property);
+		String oldValue = propertyContainer.getProperty(property).toString();
+		String newValue;
+		try(Transaction tx = db.beginTx()){
+			Lock lock = tx.acquireReadLock(propertyContainer);
+			newValue = oldValue.concat(string);
+			propertyContainer.setProperty(property, newValue);
+			tx.success();
+			lock.release();
+		}
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * insert a value into an array property value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.insert(node/relatonship,propertyName,position,value) insert a value into the property's array value at 'position'")
+	public Stream<AtomicResults> insert(@Name("container") Object container, @Name("propertyName") String property, @Name("position") Long position, @Name("value") Object value) throws ClassNotFoundException {
+		checkIsPropertyContainer(container);
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		checkPropertyExists(propertyContainer,property);
+		Object oldValue = propertyContainer.getProperty(property);
+		Object newValue;
+		try(Transaction tx = db.beginTx()){
+			Lock lock = tx.acquireReadLock(propertyContainer);
+			List<Object> values = insertValueIntoArray(oldValue, position, value);
+			Class clazz = Class.forName(values.toArray()[0].getClass().getName());
+			newValue = Array.newInstance(clazz, values.size());
+			try {
+				System.arraycopy(values.toArray(), 0, newValue, 0, values.size());
+			} catch (Exception e) {
+				String message = "Property's array value has type: " + values.toArray()[0].getClass().getName() + ", and your value to insert has type: " + value.getClass().getName();
+				throw new ArrayStoreException(message);
+			}
+			propertyContainer.setProperty(property, newValue);
+			tx.success();
+			lock.release();
+		}
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * remove a value into an array property value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.remove(node/relatonship,propertyName,position) remove the element at position 'position'")
+	public Stream<AtomicResults> remove(@Name("container") Object container, @Name("propertyName") String property, @Name("position") Long position) throws ClassNotFoundException {
+		checkIsPropertyContainer(container);
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		checkPropertyExists(propertyContainer,property);
+		Object oldValue = propertyContainer.getProperty(property);
+		Object newValue;
+		try(Transaction tx = db.beginTx()){
+			Lock lock = tx.acquireReadLock(propertyContainer);
+			Object[] arrayBackedList = new ArrayBackedList(oldValue).toArray();
```

Comment:
Just do a system.arrayCopy to idx -1 if (not first or last index) + Arrays.copyOf
array.getClass().getComponentType()